### ISSUE TITLE
Extend CSV to support fixed column size files?

### ISSUE THREAD

**Issue by dsyme** [fsharp/FSharp.Data#801]
Hi all,

I was looking at [these samples of intra-day trading prices form the NYSE](ftp://ftp.nyxdata.com/Historical%20Data%20Samples/Daily%20TAQ/) which someone pointed me to.

One of the files containing the master list of stocks contains this.  The first and second columns are fixed size, the third is variable.

I'm wondering if the CSV reader might support this case.  It's not hard to parse manually of course.

This is just a suggestion, feel free to close it pending further requests for this functionality (perhaps it's been requested before too)

```
 00846U1010000A           A              AGILENT TECHNOLOGIES, INC                                   41726005508L 000011999111820051228000000000000000010   00  A           Y114A03307920000001D119991118101  11111                                       
 0138171010000AA          AA             ALCOA INC                                                   41726005508A 000011999010420100420000000000000000010   00  AA          Y130A10696020000001D119510611101  11111                                       
 00768Y2060020AADR        AADR           ADVISORSHARES TR WCM/BNY MELLON FOCUSED GROWTH ARD ETF      400000000    000012010072120091023000000000000000010   05  AADR        N416A00002000000100B720100721000  10010                                       
...
```

**Comment by colinbull** [comment]
I think rather than extending the CSV provider, we should create a new `FixedWidth` provider which has it's own `FixedWidthFile` representation that just reuses the generator and inference from the CSV provider, to provide the types.

**Comment by tpetricek** [comment]
Supporting fixed-width column files would be nice. I guess most of the API from CSV type provider can be reused, but having a separate provider for this sounds like a good idea.

(Unless we want to write an inference that will analyze a file and check if it is fixed-width or CSV - then it would make sense to have just `WhateverMessyFormatProvider` :-))

**Comment by ovatsus** [comment]
CsvProvider is already almost whateverMessyFormalProvider. Ages ago I even suggested renaming it to something TabularTextFormatProvider. I think it would be great if this could just be autodetected, like we already autodetect tab separators instead of comma separators

**Comment by beekomann** [comment]
I don't see how the columns could be auto detected. For example here are the first 71 character columns of the US FedACH Directory.
`011001962O1210003742080312122203950CATHAY BANK                         `
The data columns are positions 1-9, 10, 11-19, 20, 21-26, 27-35, 36-71.
